app/livechat/server/lib/Livechat.js

```diff
@@ -282,6 +285,9 @@ export const Livechat = {
 			Object.assign(updateUser, { $unset: { department: 1 } });
 		} else {
 			const dep = LivechatDepartment.findOneByIdOrName(department);
+			if (!dep) {
```

Comment:
We should validate the department first, otherwise the visitor will be created but an error will be returned.